lighthouse-core/audits/byte-efficiency/unminified-javascript.js

```diff
@@ -78,20 +78,24 @@ class UnminifiedJavaScript extends ByteEfficiencyAudit {
     const items = [];
     const warnings = [];
     for (const requestId of Object.keys(artifacts.Scripts)) {
-      const scriptContent = artifacts.Scripts[requestId];
+      const scriptContentEntry = artifacts.Scripts[requestId];
+      const scriptContents = Array.isArray(scriptContentEntry) ?
+        scriptContentEntry : [scriptContentEntry];
       const networkRecord = networkRecords.find(record => record.requestId === requestId);
-      if (!networkRecord || !scriptContent) continue;
+      if (!networkRecord || !scriptContents) continue;
 
-      try {
-        const result = UnminifiedJavaScript.computeWaste(scriptContent, networkRecord);
-        // If the ratio is minimal, the file is likely already minified, so ignore it.
-        // If the total number of bytes to be saved is quite small, it's also safe to ignore.
-        if (result.wastedPercent < IGNORE_THRESHOLD_IN_PERCENT ||
-          result.wastedBytes < IGNORE_THRESHOLD_IN_BYTES ||
-          !Number.isFinite(result.wastedBytes)) continue;
-        items.push(result);
-      } catch (err) {
-        warnings.push(`Unable to process ${networkRecord.url}: ${err.message}`);
+      for (const scriptContent of scriptContents) {
+        try {
+          const result = UnminifiedJavaScript.computeWaste(scriptContent, networkRecord);
```

Comment:
I don't think this is what we want, right? IIUC, the network record is going to be the main document for all inline scripts, so we'll calculate the percent waste of a single inline script and report X% savings on the main document. Then we'll hit the next inline script and do it again?

Reply:
I'll get some reports uploaded and we'll take a look.

FYI this is the relevant code: https://github.com/GoogleChrome/lighthouse/blob/1098f952d5c75b22668aa572532415f81d2817af/lighthouse-core/audits/byte-efficiency/byte-efficiency-audit.js#L82

It should be reporting each un-minified script, and reporting some percentage relative to the main doc that can be saved.

Reply:
ACTUALLY, it reports the percentage saved relative to the script length, not the document length. fyi.

Reply:
relative to the javascript string length times the compression ratio of the main document, right?

Reply:
yup! it's very clever like that